pkgs/native_assets_builder/doc/protocol/v1.md

```diff
@@ -0,0 +1,216 @@
+Build Hook Specification
+========================
+
+Version 1: As of 2024-01-24 implemented behind `--enable-experiment=native-assets`
+
+### Concepts
+
+#### Asset
+A file that is identified by an `assetId`. There may be multiple files with the
```

Comment:
I would try to separate the structure and the properties of the structure - so `an asset is a reference to a file, with additional information such as an ID, the build target, or the way it is linked.` and then `the ID is not necessarily unique` etc.

Reply:
I took a stab at this, but I'm not entirely sure what you want here. Feel free to push commits to this branch.